src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -346,24 +346,26 @@ protected RequestExecutor(string databaseName, X509Certificate2 certificate, Doc
             UpdateConnectionLimit(initialUrls);
         }
 
-        public static RequestExecutor Create(string[] initialUrls, string databaseName, X509Certificate2 certificate, DocumentConventions conventions)
+        public static RequestExecutor Create(string[] initialUrls, string databaseName, X509Certificate2 certificate, DocumentConventions conventions, bool usePrivateUrls = false)
```

Comment:
what about ClusterRequestExecutor?
1. There are few overrides there with `new` and `[Obsolete]` that probably need to be changed now?
2. Do we need that for ClusterRequestExecutor?

Reply:
1. Please check the warnings

Reply:
I'm not sure what behavior we want regarding the `ClusterRequestExecutor`. I'm assuming it's not as straight forward as shard executors where the first priority is always private urls. Opened an issue for this https://issues.hibernatingrhinos.com/issue/RavenDB-20361/Use-private-urls-in-ClusterRequestExecutor